pool/src/lib.rs

```diff
@@ -1262,7 +1285,47 @@ impl LendingPool {
                 };
 
             if receive_stoken {
-                s_token.transfer_on_liquidation(&who, &liquidator, &s_token_amount);
+                let debt_token = DebtTokenClient::new(env, &reserve.debt_token_address);
+                let liquidator_debt = debt_token.balance(&liquidator);
+
+                if liquidator_debt == 0 {
+                    s_token.transfer_on_liquidation(&who, &liquidator, &s_token_amount);
+                } else {
+                    let debt_coeff = Self::get_debt_coeff(env, &reserve)?;
+
+                    let liquidator_actual_debt = debt_coeff
+                        .mul_int(liquidator_debt)
+                        .ok_or(Error::LiquidateMathError)?;
+
+                    let repayment_amount = liquidator_actual_debt.min(underlying_amount);
+
+                    let s_token_to_burn = coll_coeff
```

Comment:
we can do not calculate s_token_to_burn if liquidator_actual_debt == underlying_amount as it's done above and in do_repay

Reply:
my bad it's not applicable here